contracts/contracts/strategies/AerodromeEthStrategy.sol

```diff
@@ -0,0 +1,592 @@
+     * @param _weth Address of the Wrapped ETH (WETH) contract.
+     * @param _amount Amount of WETH to withdraw.
+     */
+    function withdraw(
+        address _recipient,
+        address _weth,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        require(_amount > 0, "Invalid amount");
+        require(_weth == address(weth), "Can only withdraw WETH");
+
+        emit Withdrawal(_weth, address(lpTokenAddress), _amount);
+
+        uint256 requiredLpTokens = calcTokenToBurn(_amount);
+
+        _lpWithdraw(requiredLpTokens);
+
+        /* math in requiredLpTokens should correctly calculate the amount of LP to remove
+         * in that the strategy receives enough WETH on balanced removal
+         */
+        // slither-disable-next-line unused-return
+        aeroRouterAddress.removeLiquidity(
+            address(weth),
+            address(oeth),
+            true,
+            requiredLpTokens,
+            _amount,
+            0,
+            address(this),
+            block.timestamp
+        );
+
+        // Burn all the removed OETH and any that was left in the strategy
+        uint256 oethToBurn = oeth.balanceOf(address(this));
+        IVault(vaultAddress).burnForStrategy(oethToBurn);
+
+        emit Withdrawal(address(oeth), address(lpTokenAddress), oethToBurn);
+
+        require(
+            weth.transfer(_recipient, _amount),
+            "Transfer of WETH not successful"
+        );
+    }
+
+    function calcTokenToBurn(uint256 _wethAmount)
+        internal
+        view
+        returns (uint256 lpToBurn)
+    {
+        /* The rate between coins in the pool determines the rate at which pool returns
+         * tokens when doing balanced removal (removeLiquidity call). And by knowing how much WETH
+         * we want we can determine how much of OETH we receive by removing liquidity.
+         *
+         * Because we are doing balanced removal we should be making profit when removing liquidity in a
+         * pool tilted to either side.
+         *
+         * Important: A downside is that the Strategist / Governor needs to be
+         * cognisant of not removing too much liquidity. And while the proposal to remove liquidity
+         * is being voted on the pool tilt might change so much that the proposal that has been valid while
+         * created is no longer valid.
+         */
+
+        uint256 poolWETHBalance = wethCoinIndex == 0
+            ? lpTokenAddress.reserve0()
+            : lpTokenAddress.reserve1();
+
+        /* K is multiplied by 1e36 which is used for higher precision calculation of required
+         * pool LP tokens. Without it the end value can have rounding errors up to precision of
+         * 10 digits. This way we move the decimal point by 36 places when doing the calculation
+         * and again by 36 places when we are done with it.
+         */
+        uint256 k = (1e36 * IERC20(address(lpTokenAddress)).totalSupply()) /
```

Comment:
Might want to avoid calling this 'k' because it's a very different thing than the AMM's k, which is also calculated elsewhere in the code.

Reply:
Renamed it.